src/libstore/build/derivation-goal.cc

```diff
@@ -766,6 +672,73 @@ Goal::Co DerivationGoal::tryLocalBuild() {
 }
 
 
+Goal::Co DerivationGoal::repairClosure()
```

Comment:
Maybe don't move this function? It makes the diff much bigger than needed.

Reply:
@edolstra You mean don't combine the two functions that this one sits in between?

I would have left it in place if it weren't for the fact that it is was in the way! :)

Reply:
OK I just want ahead and did it. The follow-up PR with the move and final inline is #12412.

Hopefully that makes this one extremely uncontroversial.